models/imap.go

```diff
@@ -0,0 +1,177 @@
+package models
+
+import (
+	"errors"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/glennzw/eazye"
+	log "github.com/gophish/gophish/logger"
+)
+
+// IMAP contains the attributes needed to handle logging into an IMAP server to check
+// for reported emails
+type IMAP struct {
+	UserId            int64     `json:"-" gorm:"column:user_id"`
+	Enabled           bool      `json:"enabled"`
+	Host              string    `json:"host"`
+	Port              uint16    `json:"port,string,omitempty"`
+	Username          string    `json:"username"`
+	Password          string    `json:"password"`
+	TLS               bool      `json:"tls"`
+	Folder            string    `json:"folder"`
+	RestrictDomain    string    `json:"restrict_domain"`
+	DeleteCampaign    bool      `json:"delete_campaign"`
```

Comment:
I'd consider renaming this. `DeleteCampaign` makes me think that the entire campaign will be deleted somehow. 😄

Reply:
Oops, good point! How about `DeleteCampaignEmail` or `DeleteReportedCampaign` ?

Reply:
Renamed:

```
DeleteCampaign    bool      `json:"delete_campaign"`
```

to:

```
DeleteReportedCampaignEmail bool      `json:"delete_reported_campaign_email"`
```